src/pages/MovieDetails.js

```diff
@@ -1,25 +1,62 @@
 import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import PropTypes from 'prop-types';
 
-import * as movieAPI from '../services/movieAPI';
 import { Loading } from '../components';
+import * as movieAPI from '../services/movieAPI';
 
 class MovieDetails extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      movie: '',
+      loading: true,
+    };
+    this.deleteMovie = this.deleteMovie.bind(this);
+  }
+
+  componentDidMount() {
+    movieAPI
+      .getMovie(this.props.match.params.id)
```

Comment:
Pra evitar ficar escrevendo _this.props.match.params.id_ toda hora, dá pra fazer um _destructuring_ e usá-lo nos outros lugares pra poupar espaço. Desse jeito você poderia excluir a sua função _deleteMovie_ e chamar diretamente _movieAPI.deleteMovie(id)_ no seu botão DELETAR.
```suggestion
const { id }  = this.props.match.params;
      .getMovie(id)
```